cmake/Templates/opm-project-config.cmake.in

```diff
@@ -93,6 +93,16 @@ if(NOT @opm-project_NAME@_FOUND)
 
   # this is the contents of config.h as far as our probes can tell:
 
+  # Require correct CMake standard. Needed for user modules as
+  # some software will add incompatible compile switches like
+  # -std=gnu++11 otherwise when search for (I guess because of
+  # imported targets using INTERFACE_COMPILE_FEATURES), and will
+  # break compilation because of missing c++17 features.
+  if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
this should use @CMAKE_CXX_STANDARD@ etc so we generate correct files should we move to e.g. 20 later.

Reply:
done